lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/TrinoJsonFactory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import com.fasterxml.jackson.core.TSFBuilder;
+import com.fasterxml.jackson.core.io.CharacterEscapes;
+import com.fasterxml.jackson.core.io.InputDecorator;
+import com.fasterxml.jackson.core.io.OutputDecorator;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+import static com.fasterxml.jackson.core.StreamWriteFeature.AUTO_CLOSE_TARGET;
+
+public final class TrinoJsonFactory
+        extends JsonFactory
+{
+    private static final String MUTATING_NOT_ALLOWED_MESSAGE = "TrinoJsonFactory does not allow changing configuration. Use TrinoJsonFactory.builder() instead.";
+
+    public static final JsonFactory JSON_FACTORY = new TrinoJsonFactory(builder());
```

Comment:
How do we know all mutating methods are blocked?
How do we know that, after new jackson version,  all mutating methods are blocked?

What's the benefit of sharing JSON_FACTORY on a public static field? 
At first sight it looks like unneeded complexity. Please explain.

Reply:
in other words: maybe remove this field and let everyone use `builder()`.
if someone wants to share json factory (eg for perf reasons), the caller can easily do so on eg a `private [static]` field

Reply:
I'll add a test that checks for Jackson version with the comment that if the version has changed - improve the builder/wrapping classes.

Reply:
please help me find answer to this question:

> What's the benefit of sharing JSON_FACTORY on a public static field?
At first sight it looks like unneeded complexity. Please explain.